code/game/machinery/vending.dm

```diff
@@ -136,12 +139,17 @@ var/global/num_vending_terminals = 1
 		// So not all machines speak at the exact same time.
 		// The first time this machine says something will be at slogantime + this random value,
 		// so if slogantime is 10 minutes, it will say it at somewhere between 10 and 20 minutes after the machine is crated.
-		src.last_slogan = world.time + rand(0, slogan_delay)
+		last_slogan = world.time + rand(0, slogan_delay)
 
 		power_change()
 
 	coinbox = new(src)
 
+	for(var/spath in slogan_languages)
+		if(ispath(spath))
+			slogan_languages -= spath
+			slogan_languages += new spath //convert paths into language datums
```

Comment:
Turns out this is completely unnecessary, can get the existing datum from all_languages

Will update when I get a chance